integration/commands_administration_test.go

```diff
@@ -773,6 +775,303 @@ func TestCommandsAdministrationDBStatsEmptyWithScale(t *testing.T) {
+	t.Parallel()
+
+	// rename collection must be executed from admin database.
+	adminDB := setup.SetupWithOpts(t, &setup.SetupOpts{DatabaseName: "admin"})
+
+	maxTableLen := strings.Repeat("a", 255)
+	insertCollections := []string{"foo", "buz"}
+
+	for name, tc := range map[string]struct { //nolint:vet // for readability
+		sourceCollection string
+		targetCollection string
+		to               string
+		expected         bson.D
+		err              *mongo.CommandError
+		recreateOld      bool // this indicates that the old collection should be recreated
+		targetNamespace  any  // optional, if set, ignores targetCollection and uses targetNamespace
+	}{
+		"Rename": {
+			sourceCollection: "foo",
+			targetCollection: "bar",
+			to:               "to",
+			expected:         bson.D{{"ok", float64(1)}},
+		},
+		"RenameSame": {
+			sourceCollection: "foo",
+			targetCollection: "foo",
+			to:               "to",
+			err: &mongo.CommandError{
+				Code:    20,
+				Name:    "IllegalOperation",
+				Message: `Can't rename a collection to itself`,
+			},
+		},
+		"TargetAlreadyExists": {
+			sourceCollection: "foo",
+			targetCollection: "buz",
+			to:               "to",
+			err: &mongo.CommandError{
+				Code:    48,
+				Name:    "NamespaceExists",
+				Message: `target namespace exists`,
+			},
+		},
+		"RenameDuplicate": {
+			sourceCollection: "foo",
+			targetCollection: "buz",
+			to:               "to",
+			err: &mongo.CommandError{
+				Code:    48,
+				Name:    "NamespaceExists",
+				Message: `target namespace exists`,
+			},
+		},
+		"SourceDoesNotExist": {
+			sourceCollection: "none",
+			targetCollection: "bar",
+			to:               "to",
+			err: &mongo.CommandError{
+				Code: 26,
+				Name: "NamespaceNotFound",
+				Message: "Source collection TestCommandsAdministrationRenameCollection-SourceDoesNotExist.none " +
+					"does not exist",
+			},
+		},
+		// this confirms that after we rename foo to bar and then recreate foo again,
+		// 1. the newly inserted documents exist
+		// 2. bool-false doesn't exist
+		"InsertIntoOld": {
+			sourceCollection: "foo",
+			targetCollection: "bar",
+			to:               "to",
+			expected:         bson.D{{"ok", float64(1)}},
+			recreateOld:      true,
+		},
+		"RenameMaxTableLen": {
+			sourceCollection: "foo",
+			targetCollection: maxTableLen + "a",
+			to:               "to",
+			err: &mongo.CommandError{
+				Code: 20,
+				Name: "IllegalOperation",
+				Message: "error with target namespace: Fully qualified namespace is too long. Namespace: " +
+					"TestCommandsAdministrationRenameCollection-RenameMaxTableLen.aaaaaaaaaaaaaaaaaaaaaaa" +
+					"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa" +
+					"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa" +
+					"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa Max: 255",
+			},
+		},
+		"BadParamTo": {
+			sourceCollection: "foo",
+			targetCollection: "",
```

Comment:
Do we have a test where "to" is correct, but the collection name is empty?

Reply:
@chilagrow wrote this, I believe it's because it will fail validating the missing parameter so target collection is not actually needed. Is this correct @chilagrow?

But I agree it's confusing because it's not obvious at first glance.

Reply:
Sorry not trying to blame anyone :| oops

Reply:
@rumyantseva when `to` has value
```
"EmptyTargetCollection": { sourceCollection: "foo", to: "to", targetCollection: "", }
```
that produces `to: "<dbname>."` which as far as I see it is not yet tested. So your suggested test should be added to cover missing test case for sure :hugs: 

when `to` empty like 
```
"EmptyTargetCollection": { sourceCollection: "foo", to: "", targetCollection: "", }
```
that does not add `to` field to bson. Which the test case I added.

Anyway when I was modifying originally, I wasn't sure what type of bad parameter it was testing. Maybe `to`
being something else is what we want to test?